src/compas_view2/forms/collapsiblebox.py

```diff
@@ -0,0 +1,69 @@
+from PySide2 import QtCore, QtWidgets
+
+
+class CollapsibleBox(QtWidgets.QWidget):
+    def __init__(self, title="", parent=None):
+        super(CollapsibleBox, self).__init__(parent)
+
+        self._parent = parent
+        self._expanded = False
+        self.toggle_button = QtWidgets.QToolButton(
+            text=title, checkable=True, checked=False
+        )
+        self.toggle_button.setStyleSheet("QToolButton { border: none; }")
+        self.toggle_button.setToolButtonStyle(
+            QtCore.Qt.ToolButtonTextBesideIcon
+        )
+        self.toggle_button.setIconSize(QtCore.QSize(8, 8))
+        self.toggle_button.setArrowType(QtCore.Qt.RightArrow)
+        self.toggle_button.pressed.connect(self.on_pressed)
+
+        self.content_area = QtWidgets.QScrollArea(
+            maximumHeight=0, minimumHeight=0
+        )
+        self.content_area.setSizePolicy(
+            QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Fixed
+        )
+        # self.content_area.setFrameShape(QtWidgets.QFrame.NoFrame)
+        lay = QtWidgets.QVBoxLayout(self)
+        lay.setSpacing(0)
+        lay.setContentsMargins(0, 0, 0, 0)
+        lay.addWidget(self.toggle_button)
+        lay.addWidget(self.content_area)
+
+    def update_parent(self):
+        if self._parent:
+            self._parent.expand()
+            self._parent.update_parent()
+
+    def on_pressed(self):
+        self._expanded = not self.toggle_button.isChecked()
+        if self._expanded:
+            self.expand()
+        else:
+            self.collapse()
+        self.update_parent()
+
+    def get_content_height(self):
+        return self._layout.sizeHint().height()
+
+    def setContentLayout(self, layout):
+        self._layout = layout
+        lay = self.content_area.layout()
+        del lay
```

Comment:
?

Reply:
was left over from example I looked up for, seems no use, now deleted